src/processor/queue-consumer.service.ts

```diff
@@ -106,6 +119,12 @@ export class QueueConsumerService extends WorkerHost implements OnApplicationBoo
     }
   }
 
+  @OnEvent('error.graph', { async: true, promisify: true })
+  private async handleUnknownError(error: Error) {
+    this.logger.error(`Received error.graph event: ${error.message}`);
+    await this.graphUpdateQueue.pause();
+  }
+
```

Comment:
Might want to talk to @wilwade about requirements here... this could put us into a state where the queue is permanently paused. It might be better to either:
1. Alert loudly, perhaps through a configured channel like a Slack notification, or a specific structured log
or
2. Terminate the app, so it will be detected by the provider's container orchestration system; most devops teams have alerts configured for pods failing/restarting

Reply:
Hmm...

I think 1 and 2. Alert + fail.

I also wonder if it is possible to mark the failing item and continue? Is that even worthwhile/good to do?

Reply:
This case is specific it will arise mostly due to

* Bad configuration on the service side i.e. bad schemas, or some unexpected errors
* it could also come due to bad keys etc sent to sdk
Pretty much human error in that cause the error is logged and event is emitted to pause the whole queue, given nojob will be able to recover from this error

Reply:
How about drain the queue and exit?

Reply:
@JoeCap08055 ? I'm not sure what the impact of a drain and exit would be.

Reply:
I think i can send a signal instead of exit to shutdown hooks

Reply:
I think it's more likely that an "unknown" error would be followed by more unknown errors. So draining the queue might just fail all of those jobs. Probably better to leave them pending, and just alert + crash.

Reply:
cool yeah removing the drain and we already alert, so this event listerner will just crash